RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS

suriya-prem opened the pull request:
RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS
Reason for change: Enable WAN Failover simulation in EthWAN mode
Test Procedure: Set this "Device.X_RDKCENTRAL-COM_EthernetWAN.LinkDown" TR181 param
to true for verifying the WAN failover simulation is happening or not when device is in EthWAN mode.
Risks: Low
Priority: P1

Commit:
RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS
Reason for change: Enable WAN Failover simulation in EthWAN mode
Test Procedure: Set this "Device.X_RDKCENTRAL-COM_EthernetWAN.LinkDown" TR181 param
to true for verifying the WAN failover simulation is happening or not when device is in EthWAN mode.
Risks: Low
Priority: P1
Signed-off-by: Suriyanarayanan_MP@comcast.com

suriya-prem committed:
RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS
Reason for change: Enable WAN Failover simulation in EthWAN mode
Test Procedure: Set this "Device.X_RDKCENTRAL-COM_EthernetWAN.LinkDown" TR181 param
to true for verifying the WAN failover simulation is happening or not when device is in EthWAN mode.
Risks: Low
Priority: P1

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/ethernet-agent?pullRequest=29) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/ethernet-agent?pullRequest=29) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: suriya-prem<br/>:x: smp960<br/><hr/>**smp960** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/ethernet-agent?pullRequest=29) it.</sub>

suriya-prem committed:
RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS
Reason for change: Enable WAN Failover simulation in EthWAN mode
Test Procedure: Set this "Device.X_RDKCENTRAL-COM_EthernetWAN.LinkDown" TR181 param to true for verifying the WAN failover simulation is happening or not when device is in EthWAN mode.
Risks: Low
Priority: P1

suriya-prem committed:
RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS
Reason for change: Enable WAN Failover simulation in EthWAN mode
Test Procedure: Set this "Device.X_RDKCENTRAL-COM_EthernetWAN.LinkDown" TR181 param to true for verifying the WAN failover simulation is happening or not when device is in EthWAN mode.
Risks: Low
Priority: P1

aadhithan01 committed:
Merge branch 'develop' into feature/wan_failover_sim

snayak002c committed:
Merge branch 'develop' into feature/wan_failover_sim

snayak002c committed:
Merge branch 'develop' into feature/wan_failover_sim

suriya-prem commented on source/TR-181/board_sbapi/cosa_ethernet_apis.c:
```diff
@@ -5707,4 +5723,163 @@ ANSC_STATUS CosaDmlEthPortSetPortCapability( PCOSA_DML_ETH_PORT_CONFIG pEthLink
+	pthread_cond_init(&LinkdownCond, &LinkdownAttr);
+	pthread_mutex_lock(&Linkdownlock);
+	set_time(ethAgent_Link_Status.EWanLinkDownTimeout);
+	int rv;
```
This will not occur. 
Reason: The thread creation will create only when EWanLinkDown param is set to true.. 
Incase, if we try to set the same value "true" multiple times, we won't set and throw an error since the value is already same as current value.. So, there is no chance to create multiple thread.

suriya-prem committed:
RDKB-61612 : WAN Failover Utility is required in Ethwan mode similar to DOCSIS
Reason for change: Enable WAN Failover simulation in EthWAN mode
Test Procedure: Set this "Device.X_RDKCENTRAL-COM_EthernetWAN.LinkDown" TR181 param
to true for verifying the WAN failover simulation is happening or not when device is in EthWAN mode.
Risks: Low
Priority: P1

snayak002c committed:
Merge branch 'develop' into feature/wan_failover_sim
